### PR TITLE
Add Cache Size Flags for Both Hot State And Epoch Boundaries

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Cache Flag Addition

**What does this PR do? Why is it needed?**

It adds a cache size flag both for our hot state cache and epoch boundary cache.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/flags/base.go [hunk 1]
```diff
@@ -149,4 +149,16 @@ var (
...
 	}
+	// HotStateCacheSize specifies the size of the hot state cache.
+	HotStateCacheSize = &cli.IntFlag{
+		Name:  "hot-state-cache-size",
+		Usage: "The number of states stored in our hot state cache",
+		Value: 32,
+	}
+	// EpochBoundaryCacheSize specifies the size of our epoch boundary cache.
+	EpochBoundaryCacheSize = &cli.IntFlag{
+		Name:  "boundary-cache-size",
```
Comment (terencechain) [inline beacon-chain/flags/base.go:160]:
```suggestion
		Name:  "boundary-state-cache-size",
```

**Comment by nisdas** [comment]
Marking as blocked until we see how the cluster behaves

**Comment by rauljordan** [comment]
Should we revisit @nisdas ?

**Comment by terencechain** [comment]
We shouldn't need this anymore with #7048

Instead we should define flags for `--archive, --no-validator`.. etc. Based on those, we can quantify different cache sizes

**Comment by nisdas** [comment]
Closing this as uneeded now with #7048

**Pull request closed without merging**